integration/spark/src/main/scala/org/apache/spark/sql/execution/command/view/CarbonCreateMVCommand.scala

```diff
@@ -150,10 +151,15 @@ case class CarbonCreateMVCommand(
 
   private def doCreate(session: SparkSession,
       tableIdentifier: TableIdentifier,
-      viewManager: MVManagerInSpark): MVSchema = {
+      viewManager: MVManagerInSpark,
+      viewCatalog: MVCatalogInSpark): MVSchema = {
     val logicalPlan = MVHelper.dropDummyFunction(
       MVQueryParser.getQueryPlan(queryString, session))
-    val modularPlan = checkQuery(session, logicalPlan)
+      // check if mv with same query already exists
+    if (viewCatalog.isMVWithSameQueryPresent(logicalPlan)) {
+      throw new MalformedMVCommandException("MV with same query already exists")
```

Comment:
if we can tell the user the name of the same mv, it will be better

Reply:
Handled